data/transaction/transaction.go

```diff
@@ -80,9 +80,23 @@ func (tx *Transaction) GetDataForSigning(encoder Encoder, marshalizer Marshalize
 		Options:          tx.Options,
 	}
 
+	if len(tx.GuardianAddr) > 0 {
+		ftx.GuardianAddr = encoder.Encode(tx.GuardianAddr)
+	}
+
 	return marshalizer.Marshal(ftx)
 }
 
+// HasOptionGuardianSet returns true if the guarded transaction option is set
+func (tx *Transaction) HasOptionGuardianSet() bool {
```

Comment:
some tests for L91-L99 would have been great

Reply:
done